nix/build-haskell-package.nix

```diff
@@ -22,6 +22,20 @@ let
       name = "${name}";
       inherit src;
     };
+
+  callPackageKeepDeriver = src: args:
```

Comment:
You want to add a comment explaining why this function is here.

It appears to be a copy of:

https://github.com/NixOS/nixpkgs/blob/d30e9cf3be1cc72f99d5f67772ce961b84126adf/pkgs/development/haskell-modules/make-package-set.nix#L170

Is there no way of directly using it from nixpkgs?

Reply:
Will check that.